src/write_resume_data.cpp

```diff
@@ -42,66 +42,20 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/aux_/numeric_cast.hpp"
 #include "libtorrent/torrent.hpp" // for default_piece_priority
 #include "libtorrent/aux_/numeric_cast.hpp" // for clamp
+#include "libtorrent/aux_/merkle.hpp" // for merkle_
 
 namespace libtorrent {
 
-	entry write_resume_data(add_torrent_params const& atp)
-	{
-		entry ret;
-
-		using namespace libtorrent::aux; // for write_*_endpoint()
-		ret["file-format"] = "libtorrent resume file";
-		ret["file-version"] = 1;
-		ret["libtorrent-version"] = lt::version_str;
-		ret["allocation"] = atp.storage_mode == storage_mode_allocate
-			? "allocate" : "sparse";
-
-		ret["total_uploaded"] = atp.total_uploaded;
-		ret["total_downloaded"] = atp.total_downloaded;
+namespace {
+	using write_resume_flags = flags::bitfield_flag<std::uint32_t, struct write_resume_flags_tag>;
 
-		// cast to seconds in case that internal values doesn't have ratio<1>
-		ret["active_time"] = atp.active_time;
-		ret["finished_time"] = atp.finished_time;
-		ret["seeding_time"] = atp.seeding_time;
-		ret["last_seen_complete"] = atp.last_seen_complete;
-		ret["last_download"] = atp.last_download;
-		ret["last_upload"] = atp.last_upload;
+	// only writes fields to create a .torrent file, no other resume data
+	constexpr write_resume_flags write_torrent_only = 0_bit;
```

Comment:
IMO, `_only` suffix is redundant/meaningless. It looks completely out of place in an expression like `write_torrent_only | write_resume_only`.

Reply:
perhaps these should be inverted, to mean *do* write torrent data, and *do* write resume data, and remove the `_only`

Reply:
Otherwise they shouldn't be flags:
1. Write everything
2. Write torrent only
3. Write resume data only